src/olympia/files/utils.py

```diff
@@ -563,7 +563,7 @@ def check_xpi_info(xpi_info, addon=None):
             _("Add-on ID must be 64 characters or less."))
     if not guid_optional and addon and addon.guid != guid:
         raise forms.ValidationError(_("Add-on ID doesn't match add-on."))
-    if (not addon and
+    if (not addon and guid and
```

Comment:
We validate that `guid` exists (and raise an error depending on wether or not that's allowed) above so we can safely skip here if it doesn't